GCEm/utils.py

```diff
@@ -137,6 +165,72 @@ def ensemble_collocate(ensemble, observations, member_dimension='job'):
     col_ensemble = col_members.concatenate_cube()
     return col_ensemble
 
+def LeaveOneOut(Xdata, Ydata, model='RandomForest', rndseed=0, **rf_kwargs):
+    """
+    Function to perform LeaveOneOut cross-validation with different models. 
+    """
+    from GCEm.rf_model import RFModel
+    from GCEm.gp_model import GPModel
+    from GCEm.nn_model import NNModel
+    from sklearn.linear_model import LinearRegression
+    
+    estimators = {'Linear': LinearRegression(),
+                  'RandomForest': RFModel,
+                  'GaussianProcess': GPModel,
+                  'NeuralNet': NNModel}
+    
+    if model not in estimators.keys():
+        raise Exception(f"Method needs to be one of {list(estimators.keys())}, found '{method}'.")
+    
+    # fix random seed for reproducibility 
+    # then shuffle X,Y indices so that they're not ordered (in time, for example)
+    np.random.seed(rndseed)
+    rndperm = np.random.permutation(Xdata.shape[0])
+    
+    # How many indices?
+    n_data = Xdata.shape[0]
+    
+    # Output array for test value and prediction
+    output = np.vstack([np.empty(2) for _ in range(n_data)])
+    
+    for test_idx in range(n_data):
+        """Split into training - validation sets"""
+        X_train = Xdata.iloc[rndperm[np.arange(len(rndperm))!= test_idx], :]
+        X_test  = Xdata.iloc[rndperm[test_idx], :].to_numpy().reshape(1,-1)
+        
+        Y_train = Ydata[rndperm[np.arange(len(rndperm))!= test_idx]]
+        Y_test  = Ydata[rndperm[test_idx]]
+       
+        """Construct and fit model"""
+        if model=='Linear':
+            model_ = estimators[model]
+            model_.fit(X=X_train, y=Y_train)
+            """Evaluate model on test data"""
+            predictions = model_.predict(X_test)
+
+            """Save output for validation plot  later on"""
+            output[test_idx] = (Y_test, predictions)
+            
+        else:
+            if model=='RandomForest':
+                model_ = estimators[model](training_params=X_train, 
+                                           training_data=Y_train, 
+                                           random_state=rndseed, 
+                                           **rf_kwargs)
```

Comment:
Could you change `rf_kwargs` to `model_kwargs` and then add random_state to it (if it's not already in there), that way you can lose this if/else block :-)

Reply:
Good idea!

Reply:
@duncanwp done!